### DISCUSSION TITLE
2024-03-29 - 🦙 TinaCMS - UX Roadmap - Chewing the Fat 📈

### DISCUSSION THREAD

**Discussion by sethdaily** [SSWConsulting/SSW.Rules.Content#8276] [Chewing the Fat]
Hey SSWers,

Building a roadmap for TinaCMS is a huge task with overwhelming possibilities. We think we've narrowed it down to an actionable list. What do you think?

Video: https://youtu.be/NCplHfQFOxk

Put your opinions and ideas 💡 below:

**Comment by Freego1783** [comment]
Lots of work to do! But It will be so much better after all that.

The GitHub integration for non devs looks way too painful. I think making GitHub invisible should be on top of the priority list.

**Comment by PothieuG** [comment]
I really appreciate the effort of making the road map video !
It offers us all a comprehensive overview of the project.

With the vote in the CFT, we are all involved in the big journey that awaits us!

**Comment by leahy268** [comment]
Lots of work for sure but definitely there are a few things missing

**Comment by tiagov8** [comment]
Great video.

It's too soon for specific UI recommendations, but I think we are on the right path to get TinaCMS better

**Comment by Dhruv-0987** [comment]
I think improving the media manager and user facing bugs is the number 1 priority here.

**Comment by jimmidier** [comment]
Making GitHub invisible and supporting Next.js app router should be the top 2 priorities as they're content editor and dev oriented respectively

**Comment by Aibono1225** [comment]
Good to see some AI related ideas in the roadmap. But before implementing those we need to fix the bugs raised in the video to ensure the basic functionality of TinaCMS first.

**Comment by jerryluossw** [comment]
Great video, Making GitHub invisible should be the top 1 priority, this is the core function of TinaCMS, otherwise we can't say TinaCMS is a complete system.

**Comment by KristenHu** [comment]
Making GitHub invisible is more important than any other thing, especially when developers and normal users are both adding content to the site. The designer team knows how painful it is hahaha

**Comment by AnkyChe** [comment]
Nice video! UI - Making GitHub invisible is important as it's confusing for content editors like me😖

**Comment by sylhuang** [comment]
I like the roadmap! And I would love to see us dogfooding TinaCMS for SSWRules

**Comment by drwharris** [comment]
I think that User Experience issues should be prioritised.

**Comment by babakamyljanovssw** [comment]
Nice presentation, I think the "UI - Make GitHub invisible" should be first priority.

**Comment by rubycogan** [comment]
Very excited for what TinaCMS has to offer, especially seeing it implemented on SSW Rules - which will give more visibility and feedback to the team.

**Comment by tino-liu** [comment]
I agree with fixing the UX first. That is the one that thing that is crucial for potential users.

**Comment by Calinator444** [comment]
It's always impresses me how in touch Seth is with the dev team for someone in the marketing team. SEO and whether to Gatsby or Next.js is typically the kind of thing I'd expect a dev to be focused on.

**Comment by awaltos** [comment]
I would put make GitHub invisible as a nice-to-have but the most important nice-to-have in this list. The flow is currently working although I agree with the mentioned pain points. I think that improving UI is always super important and TinaCMS could use some improvement there. Great video too! Looking forward to seeing the improvements and their impacts

**Comment by TobyChurches** [comment]
Great Video @sethdaily! Very well laid out. I personally think that 'dogfooding' the product is a great idea. If we are able to start working with it as soon as possible, we can see from the Dev perspective where the product is lacking, and highlight any great features that would promote more uptake.

**Comment by christoment** [comment]
Great video @sethdaily ! I think this early work is important for us to see where Tina place is in SSW. Ss we go along the development surely we will get more insights on how SSW want to utilise Tina, and where do we want it to go in the future 😄

**Comment by BenDunkerleySSW** [comment]
Good video Seth. Amazing to see how many great ideas we've put together for all aspects of Tina in such a short time. Very keen to see where all these plans take us in the near future.

I think UX will always be important - after all, it defines your use of the service.

**Comment by matt-goldman** [comment]
Needs a top-down focus. Start with the UX _we want_ and solution from there.

**Comment by YalalCakirSSW** [comment]
Great video Seth. I think Media manager should be the first priority. I like the "Make Github invisible" idea as well the solution seems super easy for non-dev people.

**Comment by Harry-Ross** [comment]
Really fantastic video Seth! Super great that issues that we on the website team have experienced over the past year were highlighted in such a concise way, and great that we are focusing on addressing TinaCMS community feedback.

**Comment by bahjatalaadelssw** [comment]
Good video, made all the solutions clear. The UI is important, the whole point of a editor is an easy way to manipulate the content, then future prove it with next.js or you just have an outdated and painful product.

**Comment by cvionnet** [comment]
Great job by the TinaCMS team and the presentation by Seth! 👍
Anything that makes TinaCMS easier to use TinaCMS should be prioritized. This includes tasks related to improving the UI and making GitHub invisible.

**Comment by miichaelsmedley** [comment]
Great job, I would like a vision video as well so we can see the high-level strategy as well, to design a roadmap without this makes it a little hard.

**Comment by synnoveisabel** [comment]
Great video! I think all your suggestions are going to improve TinaCMS a lot! Both fixing Media Manager and making GitHub invisible to me seems like very important tasks to boost the product 😊

**Comment by MarcusIrmscherSSW** [comment]
Nice video Seth, lots of work ahead but exciting times.

**Comment by stephanfakossw** [comment]
Great and detailed video! There's a lot of work to be done, starting with user experience.

**Comment by camillars** [comment]
Great video, Seth! There are some good recommendations there - For me, everything related to UI/UX is at the top of my suggested priorities for Tina. However, I do feel like we need to go a step back and think of what we want to achieve with the product and think of the strategy and goals for the product. 🤓

**Comment by ncn-ssw** [comment]
Voted for covering up GitHub as best as possible.

**Comment by Levijj22** [comment]
Great video Seth!
+1000 - Make Github invisible! 🤩

**Comment by jaydenalchin** [comment]
Great video but much too long for a timeboxed CTF 😉

Polishing the UX is super important not only for potential future users but for internal users (us!) as well. Really great improvements outlined by Seth here - but we should definitely zoom out & really define our goals around those target audiences/personas outlined at the start of Seth's presentation (personas that could also be further defined).

**Comment by tkapa** [comment]
Lots of things to consider with this one.

I'm excited for the prospect of polishing the UX and getting it rolled out to Rules. I think there'll be a lot of improvement on that one.

**Comment by ricksu978** [comment]
Usually I drift off in long video like this, but this is very well presented video and easy to follow.
Well Done!

**Comment by joshbermanssw** [comment]
Although the video was long and there was alot of content, Seth covered it all really well! Clear pains and inconveniences!

**Comment by JackDuan123** [comment]
My Opinion on the video:
If dream, dream big.

**Comment by vladislav-kir** [comment]
Making GitHub invisible should be the number one priority, since this is supposed to be the main purpose of CMS!

**Comment by danielmackay** [comment]
When developing content driven websites, there are two options: Static Sites, or a CMS.

Developers don't really need a CMS as they are comfortable editing markdown and code.

For Content Editors however, they need a CMS that has an easy to use and intuitive UI.  Seeing that these are the main users of a CMS making GitHub invisible is a no-brainer.

Regarding the image media manager, we need to consider that images may want to be used at different sizes in different locations.  We should add in a way to handle that so that we can optimize the image delivery (i.e. same image being used in both a website header, and full page).

I noticed a lot of the UI improvements still required content approval.  In my 7 years of working with CMS websites I've found that not all content needs approval, and some clients don't want that.  I suggest creating configurable workflows that can handle both an approval (i.e. via PR) and direct push.

**Comment by amankumarrr** [comment]
Great video! Enhancing the Media Manager's search functionality and adding a deletion feature will enable our clients to efficiently clean up unnecessary items in their Media Manager, making it easier to manage and save some cost. 

Additionally, for performance optimization, we should consider adding support for App Router as a priority.

**Comment by AntPolkanov** [comment]
Great video, watched it all!
It seems like the UI-related features are quick wins and can make life easier for editors.

**Comment by chrisschultzssw** [comment]
Great video! I agree with many other comments here - making GitHub invisible should be top priority, to make it easier to use for non-devs.

**Comment by jimmy1992abc** [comment]
I watched the whole video - it already approves the quality of itself.
Thanks my future CEO of TinaCMS

**Comment by JackDevAU** [comment]
Honestly an amazing video Seth! awesome job! 

Highlighted a lot of the issues with TinaCMS at the moment - can't wait to start working on it and fixing them :)

**Comment by ozair-ashfaque** [comment]
Great video! I look forward to these milestones being achieved, especially the Invisible GitHub.

**Comment by tombui99** [comment]
Great video from Seth which is clear that a lot of work has been put in! A lot of these works are important to have especially for content-heavy sites, this would make a much more seamless process for the editors.

**Comment by PennyWalker** [comment]
I really underestimated the time and money it would take to really get Tina underway. There are a lot of important bugs to fix as well as critical UI changes. I have a lot of confidence that we can do it, and that we know the approach from this video.

**Comment by bettybondoc** [comment]
Well done, Seth and team! Outstanding presentation of the pain points and solutions!

Making GitHub invisible definitely feels like the #1 priority. Despite my dev experience, I find using TinaCMS for website updates to be overly complex and time-consuming. Simple tasks like changing an image or text should not be so cumbersome. We won't be able to sell TinaCMS as the best CMS ever if content editors are afraid to use it.

**Comment by kikibianc** [comment]
Good list. 
I can also see people around the world commenting about our ideas in the tinacms repo e.g. https://github.com/tinacms/tinacms/discussions/4478 - that's something new for SSW, as we generally only have internal people comment on ideas. 
It's gold to see interaction outside SSW borders :)

**Comment by steven0x51** [comment]
Good video! I think to improve the user experience, the Media Manager and make github invisble would be the top 2 priority.

**Comment by Marxoz** [comment]
Great video. Really showed the pain points.

I think implementing it on SSW.Rules should be the first point, so that have something to test the rest of the changes on.

**Reply by sethdaily** [reply]
I wanted this to be first
Turned out we need to do all the prior ones before it's good enough to do that

**Comment by MattParkerDev** [comment]
I agree with the top list of priorities. Definitely on board for moving SSW Rules to Next.js

**Comment by Ben0189** [comment]
Nice video, Improving UX should be the priority

**Comment by Hona** [comment]
Some quick wins & important longer term features on the roadmap

**Comment by tiagov8** [comment]
FYI - In the past week I worked on 2 different tasks with images using Tina. I have to say that the problem that you can't find an uploaded image (at least not in an acceptable experience) it really ❌**bad**.

This should be a high priority especially because we are talking about a CMS which is supposed to make editors' life easier

**Comment by AttackOnMorty** [comment]
Nice long video. Vote for **Dogfooding - Implement TinaCMS on SSW Rules**.

We don't need to wait for everything to be perfect before using it.

**Comment by SamBWagner** [comment]
Great vid. I do with the video explained the primary consumer base so that we could've contextualised our responses. If TinaCMS is a CMS for devs, by devs, then the invisible git thing is valueless. If not, then it's super high priority.

**Comment by jeoffreyfischer** [comment]
Awesome video Seth and team!
When using TinaCMS, the media manager was the biggest issue as a Dev - so this would be my personal priority.
But as a normal user, making GitHub invisible is probably the highest priority as it would highly improve UX and make TinaCMS more valuable.

**Comment by ozdemir-mehmet-cer** [comment]
Dog fooding should be important to SSW
